src/internal/kopia/upload.go

```diff
@@ -183,6 +179,30 @@ func (cp *corsoProgress) FinishedFile(relativePath string, err error) {
 	}
 
 	cp.deets.AddFolders(folders)
+
+	// These items were sourced from a base snapshot or were cached in kopia so we
+	// never had to materialize their details in-memory.
+	if d.info == nil {
+		// TODO(ashmrtn): We should probably be returning an error here?
+		if d.prevPath == nil {
+			return
```

Comment:
Add an error log here?

Reply:
we don't have access to a context here so unless I stash a logger in the struct instance there's not a whole lot we can do :/

Reply:
The logger will fall back to a singleton instance if given the background context.  You might miss out on some logging within tests, but for the CLI it'll still write out as expected.

Reply:
thinking about this more broadly, I think we should be storing errors in the struct and returning them at the end. Otherwise we could miss the fact that we failed to add some details. I'll make a follow up ticket fo this

Reply:
tracking with #1915